src/Compilers/CSharp/Test/Syntax/Parsing/LocalFunctionParsingTests.cs

```diff
@@ -9,6 +9,73 @@ namespace Microsoft.CodeAnalysis.CSharp.UnitTests
 {
     public class LocalFunctionParsingTests : ParsingTests
     {
+        [Fact]
+        public void NoNodesWithoutExperimental()
+        {
+            // Experimental nodes should only appear when experimental are
+            // turned on in parse options
+            var file = ParseFile(@"
+class c
+{
+    void m()
+    {
+        int local() => 0;
+    }
+}");
+            Assert.NotNull(file);
+            Assert.True(file.HasErrors);
+            file.SyntaxTree.GetDiagnostics().Verify(
+                // (6,18): error CS1528: Expected ; or = (cannot specify constructor arguments in declaration)
+                //         int local() => 0;
+                Diagnostic(ErrorCode.ERR_BadVarDecl, "() => 0").WithLocation(6, 18),
+                // (6,18): error CS1003: Syntax error, '[' expected
+                //         int local() => 0;
+                Diagnostic(ErrorCode.ERR_SyntaxError, "(").WithArguments("[", "(").WithLocation(6, 18),
+                // (6,25): error CS1003: Syntax error, ']' expected
+                //         int local() => 0;
+                Diagnostic(ErrorCode.ERR_SyntaxError, ";").WithArguments("]", ";").WithLocation(6, 25));
+
+            Assert.Equal(0, file.SyntaxTree.Options.Features.Count);
+            var c = Assert.IsType<ClassDeclarationSyntax>(file.Members.Single());
+            var m = Assert.IsType<MethodDeclarationSyntax>(c.Members.Single());
+            var s1 = Assert.IsType<LocalDeclarationStatementSyntax>(m.Body.Statements[0]);
+            Assert.Equal(SyntaxKind.PredefinedType, s1.Declaration.Type.Kind());
+            Assert.Equal("int", s1.Declaration.Type.ToString());
+            var local = s1.Declaration.Variables.Single();
+            Assert.Equal("local", local.Identifier.ToString());
+            Assert.NotNull(local.ArgumentList);
+            Assert.Null(local.Initializer);
+            Assert.Equal(SyntaxKind.ParenthesizedLambdaExpression, local.ArgumentList.Arguments.Single().Expression.Kind());
+        }
+
+        [Fact]
+        public void NodesWithExperimental()
+        {
+            // Experimental nodes should only appear when experimental are
+            // turned on in parse options
+            var file = ParseFileExperimental(@"
+class c
+{
+    void m()
+    {
+        int local() => 0;
```

Comment:
Also try a {} body, please.

I suggest adding some code to assert that the tree's descendent nodes do not contain any local function declarations.